src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/SemanticTokenTestBase.cs

```diff
@@ -92,6 +109,54 @@ internal void AssertSemanticTokensMatchesBaseline(IEnumerable<int>? actualSemant
             return semanticArray;
         }
 
+        private protected async Task<ProvideSemanticTokensResponse> GetCSharpSemanticTokensResponseAsync(
+            string documentText, OS.Range razorRange, bool isRazorFile, int hostDocumentSyncVersion = 0)
+        {
+            var codeDocument = CreateCodeDocument(documentText, isRazorFile, DefaultTagHelpers);
+            var csharpRange = GetMappedCSharpRange(codeDocument, razorRange);
+            var csharpTokens = Array.Empty<int>();
+            var isFinalized = true;
+
+            if (csharpRange is not null)
+            {
+                var csharpSourceText = codeDocument.GetCSharpSourceText();
+                var (csharpLspServer, documentUri) = await CreateCSharpTestLspServerAsync(csharpSourceText, SemanticTokensServerCapabilities);
```

Comment:
I would recommend having the only return here be the IAsyncDisposable, then you can just do `await using ....` iirc, instead of having to manually call dispose async at the end.